lib/galaxy/webapps/galaxy/api/dataset_collections.py

```diff
@@ -63,14 +62,18 @@ def create(self, trans: ProvidesHistoryContext, payload: dict, **kwd):
             self.check_user_can_add_to_library_item(trans, library_folder, check_accessible=False)
             create_params["parent"] = library_folder
         else:
-            raise RequestParameterInvalidException()
+            raise exceptions.RequestParameterInvalidException()
 
         dataset_collection_instance = self.__service.create(trans=trans, **create_params)
         return dictify_dataset_collection_instance(dataset_collection_instance,
                                                    security=trans.security, parent=create_params["parent"])
 
     @expose_api
-    def show(self, trans: ProvidesHistoryContext, instance_type, id, **kwds):
+    def show(self, trans: ProvidesHistoryContext, id, instance_type='history', **kwds):
+        """
+        GET /api/dataset_collections/{hdca_id}
+        GET /api/dataset_collections/{ldca_id}?instance_type=library
```

Comment:
ldca's are not really a galaxy type, these are still hdcas, but the parent is a library.
```suggestion
        GET /api/dataset_collections/{hdca_id}?instance_type=library
```
Seems more accurate

Reply:
We do have `self.model.LibraryDatasetCollectionAssociation` in `lib/galaxy/model/__init__.py` and it seems to be used when creating a collection with `POST /api/dataset_collections` with `instance_type=library` (see also `lib/galaxy/managers/collections.py`).

Reply:
Thanks, I did not know that ...

Reply:
NP, to be fair it is not yet possible to create or add collections to libraries in the UI, so...